changelogs/unreleased/gh-8744-fix-chunked-with-content-length.md

```diff
@@ -0,0 +1,4 @@
+## bugfix/lua/http client
+
+* Fixed the `Transfer-Encoding: chunked` setting even if the `Content-Length`
+  header exists for stream requests (gh-8744).
```

Comment:
Nit: I think we should specify what is being fixed about Transfer-Encoding setting (I got quite confused even after reading the issue and curl doc). Maybe like this:
> Fixed the `Transfer-Encoding: chunked` setting being enabled even if the `Content-Length` header exists for stream requests (gh-8744).

Reply:
Thank you, updated.